client/src/components/Draftail/sources/ModalWorkflowSource.js

```diff
@@ -180,33 +195,38 @@ class ModalWorkflowSource extends Component {
   }
 
   onChosen(data) {
-    const { editorState, entityType, onComplete } = this.props;
+    const { editorState, entity, entityKey, entityType, onComplete } = this.props;
     const content = editorState.getCurrentContent();
     const selection = editorState.getSelection();
-
     const entityData = filterEntityData(entityType, data);
     const mutability = MUTABILITY[entityType.type];
-    const contentWithEntity = content.createEntity(entityType.type, mutability, entityData);
-    const entityKey = contentWithEntity.getLastCreatedEntityKey();
 
     let nextState;
-
     if (entityType.block) {
-      // Only supports adding entities at the moment, not editing existing ones.
-      // See https://github.com/springload/draftail/blob/cdc8988fe2e3ac32374317f535a5338ab97e8637/examples/sources/ImageSource.js#L44-L62.
-      // See https://github.com/springload/draftail/blob/cdc8988fe2e3ac32374317f535a5338ab97e8637/examples/sources/EmbedSource.js#L64-L91
-      nextState = AtomicBlockUtils.insertAtomicBlock(editorState, entityKey, ' ');
+      if (entity && entityKey) {
+        // Replace the data for the currently selected block
+        const blockKey = selection.getAnchorKey();
+        const block = content.getBlockForKey(blockKey);
+        nextState = DraftUtils.updateBlockEntity(editorState, block, entityData);
+      } else {
```

Comment:
I might be mistaken but it looks like there are no tests written for this new scenario in `ModalWorkflowSource.test.js`.

Reply:
Test added here
https://github.com/wagtail/wagtail/pull/5885/files#diff-36d75bb432538146bbccff1e4daffa3cR332